### PR TITLE
#2513 Fix PageBanner proportions and transparency

### LINKED ISSUE #2513: BUG: Banners stretched + displaying poorly
**Describe the bug**
Images on UN SDGS are displaying poorly:

**To Reproduce**
https://alkem.io/digileefomgeving/challenges/klimaatadaptatie/dashboard

**Expected behavior**
Image should be cropped so that the height is 1/6th of the width + then scaled to fit

**Notes from the developer**
Specifically on the image in the example I saw another problem.
The UN banner image is this: https://sdgs.un.org/themes/custom/porto/assets/goals/global-goals.png
it's a PNG with transparent background. 
That is a problem because behind the image the &lt;Skeleton&gt; makes an animation while the image loads that is supposed to be automatically hidden when the image is completely loaded. In the case of this UN banner we have this:

I have fixed it with a useState and detecting when the image loads completely to remove the Skeleton

### PR DESCRIPTION
**Pull request by ccanos**
### Describe the background of your pull request

Resolves #2513 

### Additional context

See the task, I've had to fix another problem about transparent images showing the Skelenton behind them.

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/domain/shared/components/PageHeader/PageBanner.tsx [hunk 1]
```diff
@@ -68,18 +76,28 @@ export interface PageBannerProps {
   loading?: boolean;
 }
 
-const PageBanner: FC<PageBannerProps> = ({ title, tagline, bannerUrl, showBreadcrumbs, loading = false }) => {
+const PageBanner: FC<PageBannerProps> = ({
+  title,
+  tagline,
+  bannerUrl,
+  showBreadcrumbs,
+  loading: dataLoading = false,
```
Comment (ccanos) [inline src/domain/shared/components/PageHeader/PageBanner.tsx:84]: just renamed `loading` to `dataLoading` to have that, and `imageLoading` flags

**Comment by techsmyth** [comment]
This looks to be working nicely :).

image

Testing approach:
- saved image url from one of the challenges on digileefomgeving that was not showing up properly
- used direct updateVisual query via api to modify url of the Visual for a local challenge

**Review by techsmyth — APPROVED** [review]
Nice update, also fixes how the SDGs hub displays on Demo I expect...

File: src/domain/shared/components/PageHeader/PageBanner.tsx [hunk 2]
```diff
@@ -52,14 +52,22 @@ const Ellipser = styled('div')(() => ({
...
   top: 0,
   width: '100%',
   height: '100%',
+  overflow: 'hidden',
   zIndex: 10,
 }));
 
+const Image = styled('img')(() => ({
+  objectFit: 'cover',
+  objectPosition: '0 0',
```
Comment (ccanos) [inline src/domain/shared/components/PageHeader/PageBanner.tsx:66]: @techsmyth Thanks, yes it does. That loading animation behind SDG logo shouldn't be visible anymore, and of course the image is not stretched any more.

We can play with this line, `objectPosition: '0 0',` Setting it to `50% 50%` centers the image and crops the middle, but I think cropping by the top is more consistent

**Comment by techsmyth** [comment]
@ccanos is it possible to pick the center of the image for displaying, not the top of the image?

**Comment by ccanos** [comment]
@techsmyth  A few examples:

 50% 50%:
image

0 0 (like it is at the moment)
image

50% 50%
image

0 0 
image


A vertical image: 
https://acc.alkem.io/un-sdgs/challenges/food/dashboard
https://sdgs.un.org/themes/custom/porto/assets/goals/images/img-goal-2.jpg

50% 50%
image

image



I agree, it's better 50% 50%. Fixed

File: src/domain/shared/components/PageHeader/PageBanner.tsx [hunk 3]
```diff
@@ -68,18 +76,28 @@ export interface PageBannerProps {
...
     <Root ref={containerReference}>
-      <Skeleton variant="rectangular" animation="wave" sx={{ height: '100%' }} />
-      {!loading && (
+      {imageLoading && <Skeleton variant="rectangular" animation="wave" sx={{ height: '100%' }} />}
+      {!dataLoading && (
         <>
           {showBreadcrumbs && <BreadcrumbsView />}
-          <Image src={bannerUrl} alt={`${title} - Banner image`} />
+          <ImageWrapper>
+            <Image src={bannerUrl} alt={`${title} - Banner image`} onLoad={() => setImageLoading(false)} />
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/shared/components/PageHeader/PageBanner.tsx]: Good idea, looks like this now on error:
image

**Review by techsmyth — APPROVED** [review]
@ccanos I think the middle of most images has the most relevant information, so lets go for the center alignment. If someone wants it differently then they just need to download the image and re-upload it themselves with cropping...

**Pull request merged**